stac_fastapi/pgstac/stac_fastapi/pgstac/models/links.py

```diff
@@ -0,0 +1,292 @@
+                    link.href = self.resolve(link.href)
+                    links.append(link)
+        return links
+
+
+@attr.s
+class PagingLinks(BaseLinks):
+    """Create links for paging."""
+
+    next: str = attr.ib(kw_only=True, default=None)
+    prev: str = attr.ib(kw_only=True, default=None)
+
+    def link_next(self) -> PaginationLink:
+        """Create link for next page."""
+        if self.next is not None:
+            method = self.request.method
+            if method == "GET":
+                href = merge_params(self.url, {"token": f"next:{self.next}"})
+                link = PaginationLink(
+                    rel=Relations.next,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=href,
+                )
+                return link
+            if method == "POST":
+                body = self.request.postbody
+                body["token"] = f"next:{self.next}"
+                return PaginationLink(
+                    rel=Relations.next,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=f"{self.request.url}",
+                    body=body,
+                )
+
+    def link_prev(self) -> PaginationLink:
+        """Create link for previous page."""
+        if self.prev is not None:
+            method = self.request.method
+            if method == "GET":
+                href = merge_params(self.url, {"token": f"prev:{self.prev}"})
+                return PaginationLink(
+                    rel=Relations.previous,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=href,
+                )
+            if method == "POST":
+                body = self.request.postbody
+                body["token"] = f"prev:{self.prev}"
+                return PaginationLink(
+                    rel=Relations.previous,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=f"{self.request.url}",
+                    body=body,
+                )
+
+
+@attr.s
+class CollectionLinksBase(BaseLinks):
+    """Create inferred links specific to collections."""
+
+    collection_id: str = attr.ib()
+
+    def collection_link(self, rel=Relations.collection) -> Link:
+        """Create a link to a collection."""
+        return Link(
+            rel=rel,
+            type=MimeTypes.json,
+            href=self.resolve(f"/collections/{self.collection_id}"),
+        )
+
+
+@attr.s
+class CollectionLinks(CollectionLinksBase):
+    """Create inferred links specific to collections."""
+
+    def link_self(self) -> Link:
+        """Return the self link."""
+        return self.collection_link(rel=Relations.self)
+
+    def link_parent(self) -> Link:
+        """Create the `parent` link."""
+        return Link(
+            rel=Relations.parent,
+            type=MimeTypes.json,
+            href=urljoin(self.base_url, "/"),
```

Comment:
This shouldn't be a urljoin with `/`, as this will remove any root path in the base_url. 
```suggestion
            href=self.base_url,
```

Reply:
This is actually part of a larger issue of using leading `/` in urljoins....see https://github.com/stac-utils/stac-fastapi/issues/142

Reply:
fixed in latest push